packages/editor/src/components/url-popover/index.js

```diff
@@ -0,0 +1,70 @@
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import {
+	Popover,
+	IconButton,
+} from '@wordpress/components';
+
+class URLPopover extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.toggleSettingsVisibility = this.toggleSettingsVisibility.bind( this );
+
+		this.state = {
+			isSettingsExpanded: false,
+		};
+	}
+
+	toggleSettingsVisibility() {
+		this.setState( {
+			isSettingsExpanded: ! this.state.isSettingsExpanded,
+		} );
+	}
+
+	render() {
+		const {
+			isEditing,
+			renderEditingState,
+			renderViewingState,
+			renderSettings,
+			position,
+			onClickOutside,
+		} = this.props;
+
+		const {
+			isSettingsExpanded,
+		} = this.state;
+
+		const showSettings = !! renderSettings && isSettingsExpanded;
+
+		return (
+			<Popover
+				className="editor-url-popover"
+				focusOnMount={ isEditing ? 'firstElement' : false }
+				position={ position || 'bottom center' }
+				onClickOutside={ onClickOutside }
+			>
+				<div className="editor-url-popover__row">
+					{ isEditing ? renderEditingState() : renderViewingState() }
```

Comment:
Any thoughts on having `isEditing` be entirely the responsibility of the parent component? That is, making this line:

```js
{ renderLinkInterface() }
```

Reply:
That's a good idea and solves a problem. In the media placeholder there won't be a case where isEditing is false, so something like `renderLinkInterface` allows that to be expressed in a simpler way.

Reply:
Could even use the `children` prop for a real 🔥 looking API:

```js
<URLPopover
	renderSettings={ () => (
		<div></div>
	) }
>
	{ () => (
		<div></div>
	) }
</URLPopover>
``

Reply:
nice 👍